includes/polling/os/f5.inc.php

```diff
@@ -1,20 +1,87 @@
 <?php
-$version = trim(snmp_get($device, '.1.3.6.1.4.1.3375.2.1.4.2.0', '-OQv'), '"');
-$hardware = trim(snmp_get($device, '.1.3.6.1.4.1.3375.2.1.3.5.2.0', '-OQv'), '"');
-$serial = trim(snmp_get($device, '.1.3.6.1.4.1.3375.2.1.3.3.3.0', '-OQv'), '"');
-
 use LibreNMS\RRD\RrdDefinition;
 
-$sessions = snmp_get($device, 'apmAccessStatCurrentActiveSessions.0', '-OQv', 'F5-BIGIP-APM-MIB');
+# retrieving base data for version, hardware and serial
+$mibs = 'F5-BIGIP-SYSTEM-MIB';
+$oids = [
+    'sysProductVersion.0',
+    'sysPlatformInfoMarketingName.0',
+    'sysGeneralChassisSerialNum.0',
+];
+
+$data = snmp_get_multi($device, $oids, '-OQUs', $mibs);
+$version    = $data[0]['sysProductVersion'];
+$hardware   = $data[0]['sysPlatformInfoMarketingName'];
+$serial     = $data[0]['sysProductVersion'];
+unset($data, $oids);
+
+
+# active APM sessions
+$oids['apmAccessStatCurrentActiveSessions']         = 'F5-BIGIP-APM-MIB::apmAccessStatCurrentActiveSessions.0';
```

Comment:
These arrays are undefined, you could either set them before this `$dataset = [];` or just define the arrays differently:
```php
$oids = [
    'F5-BIGIP-APM-MIB::apmAccessStatCurrentActiveSessions.0',
    'F5-BIGIP-SYSTEM-MIB::sysStatClientTotConns.0',
    'F5-BIGIP-SYSTEM-MIB::sysStatServerTotConns.0',
    'F5-BIGIP-SYSTEM-MIB::sysStatClientCurConns.0',
    'F5-BIGIP-SYSTEM-MIB::sysStatServerCurConns.0'
];

$metadata = [
    'apmAccessStatCurrentActiveSessions' => [
        'dataset' => 'sessions',
        'type' => 'GAUGE',
        'name' => 'bigip_apm_sessions'
    ],
    'sysStatClientTotConns' => [
        'dataset' => 'ClientTotConns',
        'type' => 'COUNTER',
        'name' => 'bigip_system_client_connection_rate'
    ],
    'sysStatServerTotConns' => [
        'dataset' => 'ServerTotConns',
        'type' => 'COUNTER',
        'name' => 'bigip_system_server_connection_rate'
    ],
    'sysStatClientCurConns' => [
        'dataset' => 'ClientCurConns',
        'type' => 'GAUGE',
        'name' => 'bigip_system_client_concurrent_connections'
    ],
    'sysStatServerCurConns' => [
        'dataset' => 'ServerCurConns',
        'type' => 'GAUGE',
        'name' => 'bigip_system_server_concurrent_connections'
    ],
];

$data = snmp_get_multi($device, $oids, '-OQUs');

foreach ($metadata as $key => $info) {
    $value = $data[0][$key];
    if (is_numeric($value)) {
        $rrd_def = RrdDefinition::make()->addDataset($info['dataset'], $info['type'], 0);
        $fields = [
            $info['dataset'] => $value,
        ];
        $tags = compact('rrd_def');
        data_update($device, $info['name'], $tags, $fields);
        $graphs[$info['name']] = true;
    }
}
```